### ISSUE TITLE
Speed bumps on install

### ISSUE THREAD

**Issue by thomax** [drewlyton/sanity-obsidian-plugin#2]
I'm on macOs 14.4.1, Obsidian v1.5.12.

I was thrilled to come across this plugin, just what I was looking for <3

I followed the instructions of cloning the plugin repo into my ./obsidian/plugins/ folder. The plugin appeared in Obsidian under "Community plugins" as expected, but when I toggled the enable switch, I got this error in the console:

> Plugin failure: obsidian-sanity Error: ENOENT: no such file or directory, open '/Users/SOMEPATH/.obsidian/plugins/sanity-obsidian-plugin/main.js'

I solved this by running `npm install && npm run build` in the sanity-obsidian-plugin folder. Should this happen automatically, executed by Obsidian?

After building, I found the settings UI and added API token etc. Upon testing my first edit (and hitting "Publish"), I got the following error:

> Access to XMLHttpRequest at 'https://XYZZY.api.sanity.io/v2023-05-03/data/mutate/production?returnIds=true&returnDocuments=true&visibility=sync' from origin 'app://obsidian.md' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

Which is understandable, so I went over to the Sanity project management to add a CORS origin. However, that interface (which, incidentally, I'm partly responsible for creating some years back) will not allow `app://obsidian.md` as an origin. The only solution then, as far as I can tell, is to use the wildcard * origin (which is less than ideal).

Finally, when I got Obsidian to publish a Sanity document, I realized that the markdown produced by Obsidian is sent to Sanity as `type: 'string'` not `type: 'block'`.

So to summarize, the readme could do with some more info
- npm install && npm run build
- Remember to add CORS origins in Sanity
- Heads-up regarding what sort of data types the plugin sends to Sanity

I could whip up a PR for an improved readme? Unless these mishaps I encountered are entirely my fault and could have been avoided in a better way?

Also, it would be super nice if there was an option to convert markdown to block-content/portable-text!
